src/Rules/DoctrineKeyValueStyleRule.php

```diff
@@ -164,12 +168,27 @@ public function processNode(Node $callLike, Scope $scope): array
                     continue;
                 }
 
-                // Be a bit generous here by allowing column value types that
-                // are *maybe* accepted. If we want to be strict, we can warn
-                // unless they are *definitely* accepted.
+                $argColumnType = $argColumn->getType();
                 $valueType = $argType->getValueTypes()[$keyIndex];
-                if ($argColumn->getType()->accepts($valueType, true)->no()) {
-                    $errors[] = 'Column "' . $table->getName() . '.' . $argColumnName . '" expects value type ' . $argColumn->getType()->describe(VerbosityLevel::precise()) . ', got type ' . $valueType->describe(VerbosityLevel::precise());
+
+                // Convert IntegerRangeType column types into IntegerType so
```

Comment:
I don't get why we wouldn't make use of integer ranges if present.

sure most of the time we won't get a type with a range, but if we get one, it should be compatibel wit the schema.

Reply:
The reason I've made an exception for integer ranges is because if a codebase doesn't diligently use integer ranges throughout their PHP code (i.e. they just use `int`), then this rule is practically unusable.

For example, in one of my relatively small projects, when I remove the `IntegerRangeType` transformation, I get over 1000 new errors of the kind:
> Query error: Column "message.account_id" expects value type int<0, 65535>, got type int 

It's my hope that some day I can be more strict about integer ranges, but I suspect that I'm not alone in living with a codebase that only uses `int`. In the meantime, knowing whether I have an `int` vs. `int|null` vs. `float` vs. `string` etc. is enormously valuable, and so it's my hope that we can continue to provide practical support for this situation.

Here are a couple options, but I'll do my best to accommodate whatever your preference is:
1. Skip the type transformation if the data value type is (or contains) `IntegerRangeType`, so that projects that are already using integer ranges in PHP can get more precise checks.
2. Add a configuration option to enable/disable this transformation (e.g. `enableDoctrineIntegerRangeChecks`).

Let me know what you think. Thanks!

Reply:
the arg-column type is based on our reflector inference, there will be most of the time a integer range type.

I wonder why/whether we have this very same problem in other phpstan-dba rules or whether its a already solved problem.

I agree that we shouldn't be too strict and enforce integer-ranges on the user.
in case we get a integer range from the user we should validate it though

Reply:
I've added logic to skip the transformation if the input value type is `IntegerRangeType` as well. As I note in the commit message, this doesn't handle the case where the input value is a compound type that contains an `IntegerRangeType`, but I can add that if you prefer. I was hesitant because I'm not aware of any convenient way to express "is or contains a specific type" in the PHPStan Types system. I'm still learning - thanks for your patience and guidance!

I'll look into your question about integer range issues in other rules, and will get back to you asap.

Reply:
For `SyntaxErrorInPreparedStatementMethodRule` and similar, I can't find any issues with integer ranges, because as far as I can tell there is no validation of the types apart from whether or not the database engine accepts it. This is something I have been pondering recently, as was planning to ask about soon anyway. :)

For example, if I have a table `account` with an integer column `account_id`, then the following prepared statement does not report an error, even though we're passing a string as the `account_id` instead of an int:
```php
$stmt = $db->prepare(
    'SELECT * FROM account WHERE account_id = :account_id',
    [
        'account_id' => 'foo',
    ],
);
```
Presumably this is because the database engine is perfectly content to compare `account_id` to `'foo'` even though it will never return a result. I don't know if it is even theoretically possible for phpstan-dba to detect this and report an error (it certainly seems complicated when you're doing more than just a simple equality check against a column value), but if it could, I would definitely make good use of it.

This seems a bit far afield from the current PR though, so if it's useful for you, I can submit it as a separate issue.